### PR TITLE
 Fix hangs when type invalid args at command-line

### LINKED ISSUE #5533: Missing CLI error handling in pulsar-admin
**Describe the bug**
pulsar-admin can raise Java exceptions in certain conditions

**To Reproduce**

(1) Giving an invalid command - e.g. "functions getstate" (should be "functions querystate"), with or without `--help` added

```
$ apache-pulsar-2.4.1/bin/pulsar-admin functions getstate
Expected a command, got getstate

Exception in thread "main" com.beust.jcommander.ParameterException: Asking description for unknown command: null
	at com.beust.jcommander.JCommander.getCommandDescription(JCommander.java:1003)
	at com.beust.jcommander.JCommander.usage(JCommander.java:988)
	at com.beust.jcommander.JCommander.usage(JCommander.java:980)
	at com.beust.jcommander.JCommander.usage(JCommander.java:972)
	at org.apache.pulsar.admin.cli.CmdBase.run(CmdBase.java:49)
	at org.apache.pulsar.admin.cli.PulsarAdminTool.run(PulsarAdminTool.java:218)
	at org.apache.pulsar.admin.cli.PulsarAdminTool.main(PulsarAdminTool.java:260)
```
(Hangs here, have to press `^C`)

(2) Giving a querystate command without the "--key" option

```
$ apache-pulsar-2.4.1/bin/pulsar-admin functions querystate --name womble
null

Reason: java.lang.NullPointerException: path is 'null'.
```

tcpdump shows no traffic on port 8080, so this appears to be a local CLI issue.

**Expected behavior**
Error message to be reported

### PR DESCRIPTION
**Pull request by zymap**
Fixes #5533 

*Motivation*

When we using command-line tools, it hangs after type an invalid args.

*Modifications*

We don't need to parse the command, just show the command usage.

### CODE REVIEW

**Comment by merlimat** [comment]
This change was done in #3579 because the error message was very messy otherwise. It might be good to see if there's a way to fix the current issue without completely reverting the change.

**Comment by zymap** [comment]
@merlimat Thank you for your reminder.

I use another way to fix this problem. Please take a look. Thanks.

**Comment by zymap** [comment]
run java8 tests
run integration tests

**Comment by zymap** [comment]
run java8 tests

**Comment by zymap** [comment]
run java8 tests
run integration tests

File: pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdBase.java [hunk 1]
```diff
@@ -39,14 +39,23 @@ public CmdBase(String cmdName, PulsarAdmin admin) {
         jcommander.setProgramName("pulsar-admin " + cmdName);
     }
 
+    private void tryShowCommandUsage() {
+        try {
+            String chosenCommand = jcommander.getParsedCommand();
+            jcommander.usage(chosenCommand);
+        } catch (Exception e) {
+            // it is caused by an invalid command, the invalid command can not be parsed
+            System.err.println("Invalid command, please use pulsar-admin to check out how to use");
```
Comment (sijie) [inline pulsar-client-tools/src/main/java/org/apache/pulsar/admin/cli/CmdBase.java]:
```suggestion
            System.err.println("Invalid command, please use `pulsar-admin --help` to check out how to use");
```

**Comment by zymap** [comment]
run java8 tests

**Comment by zymap** [comment]
Hi @sijie @merlimat PTAL. Thanks

**Comment by sijie** [comment]
@zymap sorry that I didn't quite get here. why did the process hang?

**Comment by zymap** [comment]
@sijie It will hang when we using command-line tools. The process hangs because there has a RuntimeException thrown and we are not handling. 

I found we exit the `PulsarAdminTool` by the `System.exit`. When we parse an unknown command it will throw a RuntimeException. We need a boolean value to check should we exit. When it threw an Exception, there is no value to return so it can not exit as usual.

**Pull request merged**